cfgrammar/src/lib/newlinecache.rs

```diff
@@ -131,6 +131,9 @@ impl NewlineCache {
             Err(j) => (self.newlines[j - 1], j),
         };
         let en = match self.newlines[st_line..].binary_search(&span.end()) {
+            Ok(j) if st_line + j == self.newlines.len() - st_line => {
+                self.newlines.last().unwrap() + self.trailing_bytes
```

Comment:
Only one minor thing, now that we have 2 of these cases, should we consider making
`input_length` public, and perhaps renaming it `input_len` (been kicking myself over that naming choice).

But i'll leave it up to you if we want to do this.

Reply:
Actually, cancel that these are in the module so, we could just use `self.input_length()`.

Reply:
I plan on doing a pass over the method names soon-ish now that the API is starting to look stable-ish. We can leave this for now though.